test/e2e/nix/build_test.go

```diff
@@ -84,7 +118,10 @@ var _ = Describe("Testing new nix implementation", func() {
 		It("running task go version will use go_1_17 from task deps", func() {
 			Expect(os.Rename("with_ambiguous_deps_in_task.yaml", "bob.yaml")).NotTo(HaveOccurred())
 
-			b, err := bob.Bob(bob.WithDir(dir), bob.WithCachingEnabled(false))
+			nix, err := bob.NewNix(bob.WithNixCache(&FakeCache{}))
```

Comment:
Can you initialize the different bob versions in BeforeSuite? It makes it harder to read this tests with this repetitive initialization.

Reply:
The thing is that bob.Bob expects a bob.yaml file with this exact name. Since each spec has a different bob.yaml I can't do that. Only if I would have different test for each spec, but not sure if that would be better in regards to readability.

I was thinking if we could add a feature to specify a custom bob file.